api/modules/files/controller.js

```diff
@@ -12,7 +12,7 @@ controller.data = function(req) {
     buffer: req.payload.buffer
   };
   if (req.params.id) {
-    data.id = req.params.id;
+    data.id = parseInt(req.params.id);
```

Comment:
Strange that you have to do this. It's an odd place for this to be forced. Why not add param validation with Joi?